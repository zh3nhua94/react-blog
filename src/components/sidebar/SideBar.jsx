import React, { useEffect, useState } from "react";
import "./sidebar.css";
import axios from "axios";
import { HashLink } from "react-router-hash-link";

const SideBar = () => {
	const PF = process.env.REACT_APP_PUBLIC_FOLDER;
	const [cats, setCats] = useState([]);

	useEffect(() => {
		const getCats = async () => {
			const res = await axios.get(process.env.REACT_APP_API_URL + "/categories");
			setCats(res.data);
		};
		getCats();
	}, []);
	return (
		<div className="sideBar">
			<div className="sideBarItem">
				<span className="sideBarTitle">ABOUT US</span>
				<img
					src={PF + "about.jpeg"}
					alt=""
					className="sideBarImg"
				/>
				<p className="sideBarText">
					Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
					magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
					commodo.
				</p>
			</div>
			<div className="sideBarItem">
				<span className="sideBarTitle">CATEGORIES</span>
				<ul className="sideBarList">
					{cats.map((c) => (
						<li
							className="sideBarListItem"
							key={c._id}
						>
							<HashLink to={`/?cat=${c.name}#home`}>{c.name}</HashLink>
						</li>
					))}
				</ul>
			</div>
			<div className="sideBarItem">
				<span className="sideBarTitle">FOLLOW US</span>
				<div className="sideBarSocial">
					<i className="sideBarIcon fa-brands fa-square-facebook"></i>
					<i className="sideBarIcon fa-brands fa-square-instagram"></i>
					<i className="sideBarIcon fa-brands fa-square-twitter"></i>
					<i className="sideBarIcon fa-brands fa-square-facebook"></i>
				</div>
			</div>
		</div>
	);
};

export default SideBar;
