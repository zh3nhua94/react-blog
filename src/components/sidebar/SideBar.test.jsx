import React from "react";
import { screen, render, waitFor, cleanup } from "@testing-library/react";
import SideBar from "./SideBar";
import axios from "axios";
import { MemoryRouter } from "react-router-dom";

jest.mock("axios");

// Mock the Axios response
const mockData = [
	{ _id: "1", name: "Category 1" },
	{ _id: "2", name: "Category 2" },
];

afterEach(cleanup);

describe("SideBar", () => {
	test("fetches and displays categories correctly", async () => {
		axios.get.mockResolvedValueOnce({ data: mockData });

		render(
			<MemoryRouter>
				<SideBar />
			</MemoryRouter>
		);

		// Wait for the Axios call to complete
		await waitFor(() => {
			// // Verify that categories are rendered
			expect(screen.getByText("Category 1")).toBeInTheDocument();
		});
		// Assertions
		expect(axios.get).toHaveBeenCalledWith(`${process.env.REACT_APP_API_URL}/categories`);
	});

	test("renders Sidebar snapshot", async () => {
		axios.get.mockResolvedValueOnce({ data: mockData });
		const { asFragment } = render(
			<MemoryRouter>
				<SideBar />
			</MemoryRouter>
		);
		await screen.findByText("Category 1");
		expect(asFragment()).toMatchSnapshot();
	});
});
