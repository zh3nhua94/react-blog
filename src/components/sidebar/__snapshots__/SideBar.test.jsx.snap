// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar renders Sidebar snapshot 1`] = `
<DocumentFragment>
  <div
    class="sideBar"
  >
    <div
      class="sideBarItem"
    >
      <span
        class="sideBarTitle"
      >
        ABOUT US
      </span>
      <img
        alt=""
        class="sideBarImg"
        src="http://localhost:8800/images/about.jpeg"
      />
      <p
        class="sideBarText"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo.
      </p>
    </div>
    <div
      class="sideBarItem"
    >
      <span
        class="sideBarTitle"
      >
        CATEGORIES
      </span>
      <ul
        class="sideBarList"
      >
        <li
          class="sideBarListItem"
        >
          <a
            href="/?cat=Category 1#home"
          >
            Category 1
          </a>
        </li>
        <li
          class="sideBarListItem"
        >
          <a
            href="/?cat=Category 2#home"
          >
            Category 2
          </a>
        </li>
      </ul>
    </div>
    <div
      class="sideBarItem"
    >
      <span
        class="sideBarTitle"
      >
        FOLLOW US
      </span>
      <div
        class="sideBarSocial"
      >
        <i
          class="sideBarIcon fa-brands fa-square-facebook"
        />
        <i
          class="sideBarIcon fa-brands fa-square-instagram"
        />
        <i
          class="sideBarIcon fa-brands fa-square-twitter"
        />
        <i
          class="sideBarIcon fa-brands fa-square-facebook"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
